cgi/mapcache.c

```diff
@@ -228,6 +242,43 @@ static void load_config(mapcache_context *ctx, char *filename)
 
 }
 
+static void set_headers(mapcache_context* ctx, char** env)
+{
+    // add all environ settings including HTTP headers to
+    // a ctx->headers_in apr_table_t
+
+    char* key, * val, * kvp, * pair;
+    int i;
+    int num_env_var;
+    apr_table_t* headers;
+
+    num_env_var = 0;
+    while (env[num_env_var] != NULL)
+        num_env_var++;
+
+    headers = apr_table_make(ctx->pool, num_env_var);
+
+    for (i = 0; env[i] != NULL; i++) {
+        kvp = env[i];
+
+        // convert HTTP header keys from the form HTTP_MY_HEADER to MY-HEADER
+        key = apr_strtok(kvp, "=", &pair);
+        key = mapcache_util_str_replace(ctx->pool, key, "HTTP_", "");
+        key = mapcache_util_str_replace(ctx->pool, key, "_", "-");
+        // key = str_replace_all(ctx->pool, key, "_", "-");
```

Comment:
Congratulations! you've found a bug in `str_replace_all()`! For solving it, you may want to use `mapcache_util_str_replace_all()` in `util.c` instead.